sdk/aqueduct/artifacts/generic_artifact.py

```diff
@@ -24,20 +23,19 @@ def __init__(
         content: Optional[Any] = None,
         from_flow_run: bool = False,
     ):
+        if content is not None:
+            assert artifact_type != ArtifactType.UNTYPED
+
```

Comment:
Is `artifact_type` only needed for this assertion purpose?

Reply:
yep I think so. Added a comment to make it more clear.